pkg/apis/core/validation/validation.go

```diff
@@ -2006,11 +2009,27 @@ func ValidatePersistentVolumeUpdate(newPv, oldPv *core.PersistentVolume, opts Pe
 }
 
 // ValidatePersistentVolumeStatusUpdate tests to see if the status update is legal for an end user to make.
-func ValidatePersistentVolumeStatusUpdate(newPv, oldPv *core.PersistentVolume) field.ErrorList {
+func ValidatePersistentVolumeStatusUpdate(newPv, oldPv *core.PersistentVolume, opts PersistentVolumeValidationOptions) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPv.ObjectMeta, &oldPv.ObjectMeta, field.NewPath("metadata"))
 	if len(newPv.ResourceVersion) == 0 {
 		allErrs = append(allErrs, field.Required(field.NewPath("resourceVersion"), ""))
 	}
+
+	if opts.EnablePersistentVolumeLastPhaseTransitionTime {
+		allErrs = append(allErrs, validateLastPhaseTransition(&newPv.Status, &oldPv.Status)...)
+	}
+
+	return allErrs
+}
+
+func validateLastPhaseTransition(newPvStatus, oldPvStatus *core.PersistentVolumeStatus) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if oldPvStatus.LastPhaseTransitionTime.IsZero() || newPvStatus.LastPhaseTransitionTime.IsZero() {
```

Comment:
According to [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/3762-persistent-volume-last-phase-transition-time/README.md#design-details), it should remove the timestamp in LastPhaseTransitionTime field during volume status update when feature gate is disabled. 

If it's allowed, we need add feature-gate check in validation. Otherwise, kep should be updated.

Reply:
@carlory Opened KEP update PR to match it with changes discussed here: https://github.com/kubernetes/enhancements/pull/4125